kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/matchers.kt

```diff
@@ -123,18 +123,15 @@ fun <T, C : Collection<T>> contain(t: T) = object : Matcher<C> {
    )
 }
 
-infix fun <T> Array<T>.shouldNotContainExactly(expected: Array<T>) = asList().shouldNotContainExactly(expected.asList())
+fun <T> Array<T>?.shouldNotContainExactly(vararg expected: T) = this?.asList() shouldNot containExactly(*expected)
```

Comment:
I wanted to, but the method signitures conflict since `expected: Array<T>` and `vararg expected: T` both resolve to an array parameter. Decided to go with the vararg since an array can simply still be passed in using the spread operator.

If we want both then one of the method will need a different name using `@JvmName` I believe.